### ISSUE TITLE
[pigeon] Codegen can accidentally use reserved keywords

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#52892]
When upgrading video_player to pigeon I ran into 2 different issues where codegen was using symbols which had reserved semantics in the target language:

- A message called "init" - when codegening for the message in objc "init:error:" was generated which has special rules in objc (it should return instancetype, which it wasn't)
- A field called "package" - the java parser is dumb and will get tripped up if you have a field that is named package, reserving that word for top level statements.

We can try to build these rules into the pigeon parser as we find them.

cc @xster

**Comment by xster** [comment]
Agree. Let's just block the superset of all reserved keywords across all languages in the IDL and tools.

**Comment by gnprice** [comment]
On the later issue #88545, @gaaclarke added:
> We can either block them from existing in the front end parser or mangle then in the output for languages where it makes a difference.
